test/suite/test_layered37.py

```diff
@@ -0,0 +1,120 @@
+    conn_base_config = ',create,cache_size=10GB,statistics=(all),statistics_log=(wait=1,json=true,on_close=true),' \
+                 + 'disaggregated=(page_log=palm),'
+
+    disagg_storages = gen_disagg_storages('test_layered37', disagg_only = True)
+
+    scenarios = make_scenarios(disagg_storages)
+
+    uri = 'layered:test_layered37'
+
+    nitems = 20000
+
+    def conn_config(self):
+        return self.extensionsConfig() + self.conn_base_config + 'disaggregated=(role="leader")'
+
+    def test_ping_ingest_table(self):
+        # Create the oplog
+        oplog = Oplog()
+
+        # Create the table on leader and tell oplog about it
+        self.session.create(self.uri, "allocation_size=512,leaf_page_max=512,key_format=S,value_format=S")
+        t = oplog.add_uri(self.uri)
+
+        # Create the follower and create its table
+        # To keep this test relatively easy, we're only using a single URI.
+        conn_follow = self.wiredtiger_open('follower', self.extensionsConfig() + self.conn_base_config + 'disaggregated=(role="follower")')
+        session_follow = conn_follow.open_session('')
+        session_follow.create(self.uri, "allocation_size=512,leaf_page_max=512,key_format=S,value_format=S")
+
+        # Load the data for oplog
+        oplog.insert(t, self.nitems)
+
+        # Apply them to leader WT.
+        oplog.apply(self, self.session, 0, self.nitems)
+        oplog.check(self, self.session, 0, self.nitems)
+
+        ts = oplog.last_timestamp()
+
+        self.conn.set_timestamp(f'stable_timestamp={self.timestamp_str(ts)}')
+
+        # On the follower -
+        # Apply all the entries to follower
+        oplog.apply(self, session_follow, 0, self.nitems)
+        oplog.check(self, session_follow, 0, self.nitems)
+
+        # Ping a read transaction
+        session_follow2 = conn_follow.open_session('')
+        session_follow2.begin_transaction(f'read_timestamp={self.timestamp_str(ts)}')
+        cursor = session_follow2.open_cursor(self.uri)
+        # Ping the cursor on the ingest table
+        self.assertEqual(cursor.next(), 0)
+
+        # Remove all data
+        oplog.remove(t, self.nitems)
+
+        # Apply them to leader WT and checkpoint.
+        oplog.apply(self, self.session, self.nitems, self.nitems)
+        oplog.check(self, self.session, self.nitems, self.nitems)
+
+        # Make all the data obsolete
+        self.conn.set_timestamp(f'stable_timestamp={self.timestamp_str(oplog.last_timestamp())},oldest_timestamp={self.timestamp_str(oplog.last_timestamp())}')
+
+        self.session.checkpoint()
+
+        # Apply them to follower.
+        oplog.apply(self, session_follow, self.nitems, self.nitems)
+        oplog.check(self, session_follow, self.nitems, self.nitems)
+
+        # Then advance the checkpoint and make sure everything is still good
+        self.disagg_advance_checkpoint(conn_follow)
+
+        # Trigger eviction on the ingest table
```

Comment:
The test is not triggering the code we want to test. The current problem is that it seems that all the data on the ingest table are on the single page. As we ping the page before, so this page cannot be evicted. I have tried to use a small page size but it doesn't help. Increasing the number of items simply make the test hang. Not sure if we can write a python test to validate this. @ddanderson

Reply:
@quchenhao, I added `test_layered38.py`, it tests this in a different way.  I ran `test_layered37.py` with larger data sizes for a while (something I just added to the Oplog class), and I see the hang in the debugger, but I don't yet understand it.  I think if you're okay with this, we can merge this (including existing test_layered37.py) and open another ticket to investigate the layered37 hanging problem.

Reply:
sgtm!